src/storm/templates/_helpers.tpl

```diff
@@ -75,26 +75,53 @@ Create a fully qualified zookeeper name.
 We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
 */}}
 {{- define "storm.zookeeper.fullname" -}}
-{{- $name := default .Values.zookeeper.service.name -}}
-{{- printf "%s-%s" (include "storm.fullname" .) $name | trunc 63 | trimSuffix "-" -}}
+{{- if .Values.zookeeper.fullnameOverride -}}
+{{- .Values.zookeeper.fullnameOverride | trunc 63 | trimSuffix "-" -}}
+{{- else -}}
+{{- $name := default "zookeeper" .Values.zookeeper.nameOverride -}}
+{{- printf "%s-%s" .Release.Name $name | trunc 63 | trimSuffix "-" -}}
+{{- end -}}
+{{- end -}}
+
+{{- define "storm.zookeeper.client.port" -}}
+{{- default 2181 .Values.zookeeper.service.ports.client.port -}}
 {{- end -}}
 
 {{- define "storm.logging.name" -}}
 {{- printf "%s-logging" (include "storm.fullname" .) | trunc 63 | trimSuffix "-" -}}
 {{- end -}}
 
-{{/*
-Override the zookeeper service name for the zookeeper chart so that both charts reference the same zookeeper service name.
-*/}}
-{{- define "zookeeper.fullname" -}}
-{{- if .Values.fullnameOverride -}}
-{{- .Values.fullnameOverride | trunc 63 | trimSuffix "-" -}}
+# TODO: There must be a better way than duplicating this block twice.
```

Comment:
possible https://godoc.org/text/template#hdr-Nested_template_definitions

Reply:
Yep, that's one of the possible solutions I was mulling.  In an upcoming version of this same code, it would save me a whole two lines of code... so I haven't pulled the trigger on refactoring it yet.